code/modules/background/origins/origins/ipc/megacorporate.dm

```diff
@@ -0,0 +1,48 @@
+	possible_accents = list(ACCENT_SILVERSUN_EXPATRIATE, ACCENT_SILVERSUN_ORIGINAL, ACCENT_SOL, ACCENT_ERIDANI, ACCENT_LUNA, ACCENT_EARTH, ACCENT_VENUS, ACCENT_CETI)
+	possible_citizenships = list(CITIZENSHIP_NONE)
+	possible_religions = list(RELIGION_NONE)
+
+/decl/origin_item/origin/hephaestus
+	name = "Hephaestus Industries"
+	desc = "Hephaestus Industries is known as the primary fabricator of IPC in the Spur. Its own lines however focus mainly on strength and mining, the company fielding untold numbers of industrial frames for all sorts of excavation, cargo, and  engineering operations, gaining them a sturdy reputation of durability and efficiency. The company is known to treat its IPCs fairly, albeit holding them to a higher standard than organics, and even offers synthetics who manage to purchase their freedom respectable positions within the company."
+	possible_accents = list(ACCENT_SOL, ACCENT_EARTH, ACCENT_MARTIAN, ACCENT_CETI, ACCENT_COC, ACCENT_NCF, ACCENT_PHONG, ACCENT_ERIDANI, ACCENT_VALKYRIE, ACCENT_GIBSON)
+	possible_citizenships = list(CITIZENSHIP_NONE)
+	possible_religions = list(RELIGION_NONE)
+
+/decl/origin_item/origin/zenghu
+	name = "Zeng-Hu Pharmaceuticals"
+	desc = "Zeng-Hu Pharmaceuticals employs a large number of IPC, both free and owned, with the newest models often being shown to the public as a display of the company's technological prowess. They are treated relatively well based on the standards of the Spur, however they are often urged to perfectionism and quality, being rewarded for success and punished for failure. In keeping with the company's competitive yet meritocratic nature, IPC which perform well and catch the eye of their superiors can be elevated to positions of responsibility, in some cases, winning their freedom through hard and exceptional labour."
+	important_information = "IPC belonging to Zeng-Hu do not possess a standard naming convention."
+	possible_accents = list(ACCENT_SOL, ACCENT_LUNA, ACCENT_CETI, ACCENT_KONYAN, ACCENT_EUROPA, ACCENT_ERIDANI)
+	possible_citizenships = list(CITIZENSHIP_NONE)
+	possible_religions = list(RELIGION_NONE)
+
+/decl/origin_item/origin/zavodskoi
+	name = "Zavodskoi Interstellar" //here lies necropolis, died a slow and painful death, thanks god
+	desc = "Zavodskoi Interstellar fields a multitude of IPC oriented towards combat and security solutions. From Industrials to Shells, these units are regularly leased out to third parties as security guards, representing the company in a professional and clean manner. Standards are high for Zavodskoi synthetics, with regular mind wiping and retooling in order to enforce discipline and minimise deviancy."
+	important_information = "Units from Zavodskoi Interstellar are all given the prefix Z.I. in front of their name."
```

Comment:
A bit out of scope, as this would require a bunch more code to deal with this SQL-side as well.

Reply:
You mention naming in the PR description though. It isn't super important especially if it would only be used for one type of ipc.

Reply:
I was only going to go as far as putting the naming requirements in the game in red text so that people functionally cannot miss it. We can do it programmatically when it's as simple as ZIs having to have the Z.I. prefix, but when it comes to Gadpathur requiring indian/pakistani/nepali names, it gets a lot more complex. May as well enforce it all administratively.

The point is, this'll make our jobs a lot easier by cutting down on most of the cases, hopefully.